### PR TITLE
[JS] feat: Updates to OpenAIModel and OpenAIEmbeddings classes.

### PR DESCRIPTION
**Pull request by Stevenic**
## Linked issues

closes: #1201 

## Details

These changes update the `OpenAIModel` and `OpenAIEmbeddings` classes to support recent parameter changes by OpenAI. They also add support for calling other OpenAI compliant services like [LLaMA.cpp](https://github.com/ggerganov/llama.cpp/blob/master/examples/server/README.md).

#### Change details

- Added new `OpenAILikeModelOptions` interface and updated `OpenAIModel` class to conditionally send an Authorization header to the model server.
- Updated `BaseOpenAIModelOptions` interface and `OpenAIModel` class to support OpenAI's `response_format` parameter.
- Updated `BaseOpenAIModelOptions` interface and `OpenAIModel` class to support OpenAI's `seed` parameter.
- Added new `OpenAILikeEmbeddingOptions` interface and updated `OpenAIEmbeddingl` class to conditionally send an Authorization header to the model server.
- Updated `BaseOpenAIEmbeddingOptions` interface and `OpenAIEmbeddings` class to support OpenAI's new `dimension` parameter.
- Tweaked documentation comment for embeddings class.

## Attestation Checklist

- [ x] My code follows the style guidelines of this project

- I have checked for/fixed spelling, linting, and other errors
- I have commented my code for clarity
- I have made corresponding changes to the documentation (updating the doc strings in the code is sufficient)
- My changes generate no new warnings
- I have added tests that validates my changes, and provides sufficient test coverage. I have tested with:
  - Local testing
  - E2E testing in Teams
- New and existing unit tests pass locally with my changes

### CODE REVIEW

File: js/packages/teams-ai/src/embeddings/OpenAIEmbeddings.ts [hunk 1]
```diff
@@ -66,6 +71,35 @@ export interface OpenAIEmbeddingsOptions extends BaseOpenAIEmbeddingsOptions {
     endpoint?: string;
 }
 
+/**
+ * Options for configuring an embeddings object that calls and `OpenAI` complient endpoint.
```
Comment (corinagum) [inline js/packages/teams-ai/src/embeddings/OpenAIEmbeddings.ts:75]:
```suggestion
 * Options for configuring an embeddings object that calls an `OpenAI` complient endpoint.
```

File: js/packages/teams-ai/src/embeddings/OpenAIEmbeddings.ts [hunk 2]
```diff
@@ -66,6 +71,35 @@ export interface OpenAIEmbeddingsOptions extends BaseOpenAIEmbeddingsOptions {
...
+ * @remarks
+ * The endpoint should comply with the OpenAPI spec for OpenAI's API:
+ * 
+ * https://github.com/openai/openai-openapi
+ * 
+ * And an example of a complient endpoint is LLaMA.cpp's reference server:
```
Comment (corinagum) [inline js/packages/teams-ai/src/embeddings/OpenAIEmbeddings.ts:81]:
```suggestion
 * And an example of a compliant endpoint is LLaMA.cpp's reference server:
```

File: js/packages/teams-ai/src/models/OpenAIModel.ts [hunk 3]
```diff
@@ -83,6 +98,35 @@ export interface OpenAIModelOptions extends BaseOpenAIModelOptions {
     endpoint?: string;
 }
 
+/**
+ * Options for configuring a model that calls and `OpenAI` complient endpoint.
```
Comment (corinagum) [inline js/packages/teams-ai/src/models/OpenAIModel.ts:102]:
```suggestion
 * Options for configuring a model that calls and `OpenAI` compliant endpoint.
```

File: js/packages/teams-ai/src/models/OpenAIModel.ts [hunk 4]
```diff
@@ -83,6 +98,35 @@ export interface OpenAIModelOptions extends BaseOpenAIModelOptions {
...
+ * @remarks
+ * The endpoint should comply with the OpenAPI spec for OpenAI's API:
+ * 
+ * https://github.com/openai/openai-openapi
+ * 
+ * And an example of a complient endpoint is LLaMA.cpp's reference server:
```
Comment (corinagum) [inline js/packages/teams-ai/src/models/OpenAIModel.ts:108]:
```suggestion
 * And an example of a compliant endpoint is LLaMA.cpp's reference server:
```

File: js/packages/teams-ai/src/models/OpenAIModel.ts [hunk 5]
```diff
@@ -120,13 +164,13 @@ export class OpenAIModel implements PromptCompletionModel {
...
      */
-    public readonly options: OpenAIModelOptions | AzureOpenAIModelOptions;
+    public readonly options: OpenAIModelOptions | AzureOpenAIModelOptions | OpenAILikeModelOptions;
 
     /**
      * Creates a new `OpenAIModel` instance.
      * @param {OpenAIModelOptions} options - Options for configuring the model client.
      */
-    public constructor(options: OpenAIModelOptions | AzureOpenAIModelOptions) {
+    public constructor(options: OpenAIModelOptions | AzureOpenAIModelOptions | OpenAILikeModelOptions) {
```
Comment (corinagum) [inline js/packages/teams-ai/src/models/OpenAIModel.ts:173]: Does `OpenAILikeModelOptions` also apply to Azure AI Studio? What is the extent of its applicability?

Comment (Stevenic) [reply js/packages/teams-ai/src/models/OpenAIModel.ts:173]: We don't need a separate one... In AlphaWave We called it out as `OpenSourceModelOptions` but OpenAILike is actually a better descriptor as it has little to do with OSS based models and is more about conforming to OpenAI's REST contract.

**Review by corinagum — APPROVED** [review]

**Pull request merged**